### PR TITLE
kebabcase with more tests than in the official documentation

### PR DESCRIPTION
**Pull request by doydoy44**

### CODE REVIEW

**Comment by doydoy44** [comment]
@102 I change target and resolve conflict ^^ (cf. https://github.com/cedmax/youmightnotneed/pull/207)

**Comment by doydoy44** [comment]
Codacy Static Code Analysis failed.
tests are run locally and tests run on old PR (https://github.com/cedmax/youmightnotneed/pull/207)
the retry return 404.
does anyone have a solution ?

**Comment by 102** [comment]
> @102 I change target and resolve conflict ^^ (cf. #207)

you can change the target branch in the PR without creating a new one

> Codacy Static Code Analysis failed. tests are run locally and tests run on old PR (#207) the retry return 404. does anyone have a solution ?

I don't really have an access to infra, so we can only wait for the feedback at the moment 🙂

**Comment by cedmax** [comment]
Not sure what's wrong with codacy, it sometimes throws the most random errors. Also, I'm off work this week and travelling, so I might look into that only in a few days 😅😅

File: src/content/lodash/string/kebabCase/lodash.js
Comment (cedmax) [inline src/content/lodash/string/kebabCase/lodash.js]: do you mind following the pattern `exports.varname`, instead of `module.exports.varname`? on the top of my mind, I'm not sure if `module.exports` gets correctly replaced in the UI, but also for consistency reasons. It should work all the same afaik

File: src/content/lodash/string/kebabCase/vanilla.js
Comment (cedmax) [inline src/content/lodash/string/kebabCase/vanilla.js:2]: Given that this regex is way more complicated than the chunks in my previous version, we need some comments explaining what it's doing. Also the notes.md file should have a reference to the negative lookahead (as well as the `??`)

File: src/content/lodash/string/kebabCase/lodash.js
Comment (doydoy44) [reply src/content/lodash/string/kebabCase/lodash.js]: copy/paste Error ^^

File: src/content/lodash/string/kebabCase/vanilla.js
Comment (doydoy44) [reply src/content/lodash/string/kebabCase/vanilla.js:2]: I have added explanations in the `notes.md` file

English is not my strong point, so don't hesitate to correct me

Comment (cedmax) [reply src/content/lodash/string/kebabCase/vanilla.js:2]: We need to see how it gets printed in the site as the notes were not really meant to be extensive descriptions to begin with and I'm not sure the way I consume them makes sense with this use case. 

But if that's all right with you, I'd also be happy to merge your PR into my branch when you are happy with it and tweak the notes (moving them to comments in code, if needed for example) before merging to the production branch

Comment (doydoy44) [reply src/content/lodash/string/kebabCase/vanilla.js:2]: Do your best, the goal is to share and improve :)

File: src/content/lodash/string/kebabCase/vanilla.js
Comment (cedmax) [inline src/content/lodash/string/kebabCase/vanilla.js]: could you do the same here with the `module.exports.varname` to `exports.varname`, please? After that happy to merge

Comment (doydoy44) [reply src/content/lodash/string/kebabCase/vanilla.js]: Of course ^^

**Review by cedmax — APPROVED** [review]

**Pull request merged**
